src/components/Requirements/Requirements.vue

```diff
@@ -169,22 +194,47 @@ export default Vue.extend({
     },
     onShowAllCourses(showAllCourses: {
       requirementName: string;
-      subReqCoursesArray: SubReqCourseSlot[];
+      subReqCoursesArray: FirestoreSemesterCourse[];
     }) {
       this.shouldShowAllCourses = true;
-      const allPotentialCourses = showAllCourses.subReqCoursesArray
-        .flatMap(slot => (slot.isCompleted ? [] : slot.courses))
-        .slice(0, 24);
-      const lastCourse = allPotentialCourses[allPotentialCourses.length - 1];
-      this.lastLoadedShowAllCourseId = lastCourse.crseId;
+
       this.showAllCourses = {
         name: showAllCourses.requirementName,
-        courses: allPotentialCourses,
+        shownCourses: this.findPotentialSeeAllCourses(showAllCourses.subReqCoursesArray),
+        allCourses: showAllCourses.subReqCoursesArray,
       };
     },
+    nextPage() {
+      if (!this.hasNextPage) {
+        return;
+      }
+
+      this.showAllPage += 1;
+      this.showAllCourses.shownCourses = this.findPotentialSeeAllCourses(
+        this.showAllCourses.allCourses
+      );
+    },
+    prevPage() {
+      if (!this.hasPrevPage) {
+        return;
+      }
+
+      this.showAllPage -= 1;
+      this.showAllCourses.shownCourses = this.findPotentialSeeAllCourses(
+        this.showAllCourses.allCourses
+      );
+    },
+    // return an array consisting of the courses to display on the see all menu, depending on the showAllPage and maxSeeAllCoursesPerPage
+    findPotentialSeeAllCourses(courses: FirestoreSemesterCourse[]): FirestoreSemesterCourse[] {
+      const allPotentialCourses = courses.slice(
+        this.showAllPage * this.maxSeeAllCoursesPerPage,
+        (this.showAllPage + 1) * this.maxSeeAllCoursesPerPage
+      );
+      return allPotentialCourses;
+    },
     backFromSeeAll() {
       this.shouldShowAllCourses = false;
-      this.showAllCourses = { name: '', courses: [] };
+      this.showAllCourses = { name: '', shownCourses: [], allCourses: [] };
     },
```

Comment:
In `backFromSeeAll()` can we also reset `this.showAllPage = 0`? Currently, if I flip to the 2nd page for the See All panel of a requirement, if I click on the See All for another requirement, I automatically get directed to the 2nd page.

https://user-images.githubusercontent.com/54298311/109899264-98143600-7c63-11eb-9db7-2d394b0feabb.mov

This could be bad in the case where the page (i.e. 2nd page) doesn't exist in the See All panel of a req:
![Screen Shot 2021-03-03 at 9 01 52 PM](https://user-images.githubusercontent.com/54298311/109899325-afebba00-7c63-11eb-821b-70a5afc5734c.png)